### ISSUE TITLE
Add eval_set to fit() to fit() method

### ISSUE THREAD

**Issue by pivettamarcos** [sebp/scikit-survival#466]
It would be nice if the fit() method for the models had eval_set parameter. This way I could define the validation set I want to use for early stopping, and not use the holdout set currently defined by scikit-survival.

**Comment by sebp** [comment]
Welcome and thanks for your interest in scikit-survival.

Can you please provide a more detailed description of what you are 
looking for? Which classifier do you want to apply early stopping to? 
What do you exactly mean by "holdout set currently defined" by 
scikit-survival"?

**Comment by tpnk-dev** [comment]
I'm currently using GradientBoostingSurvivalAnalysis. Would it be possible to implement a parameter like "eval_set" passed to the fit() function just like it's possible with the XGBoost package?

I'm aware of the validation_fraction parameter, but I have no control over what portion of data will be used for validation.

**Comment by pivettamarcos** [comment]
I'm currently using GradientBoostingSurvivalAnalysis. Would it be possible to implement a parameter like "eval_set" passed to the fit() function just like it's possible with the XGBoost package?

I'm aware of the validation_fraction parameter, but I have no control over what portion of data will be used for validation.

**Comment by sebp** [comment]
If you want full control, use the [monitor argument of fit()](https://scikit-survival.readthedocs.io/en/stable/api/generated/sksurv.ensemble.GradientBoostingSurvivalAnalysis.html#sksurv.ensemble.GradientBoostingSurvivalAnalysis.fit).
